repair/repair.cc

```diff
@@ -1478,3 +1491,52 @@ future<> repair_abort_all(seastar::sharded<database>& db) {
         repair_tracker().abort_all_repairs();
     });
 }
+
+future<> sync_data_with_repair(seastar::sharded<database>& db,
+        sstring keyspace,
+        dht::token_range_vector ranges,
+        std::unordered_map<dht::token_range, repair_neighbors> neighbors) {
```

Comment:
Why unordered_map and not a vector? We're not looking up anything.

Reply:
We look up token_range for the repair_neighbors. We manipulate ranges in the caller site. If we only pass the a vector of ranges and a vector of repair_neighbors. We might have the wrong ordering between the ranges and the repair_neighbors. Using a map is safer because the order does not matter.